winrm/x509.go

```diff
@@ -0,0 +1,198 @@
+// Copyright 2016 Canonical Ltd.
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package winrm
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/juju/errors"
+	"github.com/juju/utils"
+	"github.com/juju/utils/cert"
+)
+
+// List of exported internal errors
+var (
+	errNoClientCert       = errors.New("No client winrm cert inside dir")
+	errNoClientPrivateKey = errors.New("No client private key inside dir")
+	errNoX509Folder       = errors.New("No configuration x509 folder is present in this location")
+)
+
+// X509 type that defines windows remote manager
+// credentials for client-server secure communication
+type X509 struct {
+	mu sync.Mutex
+
+	cert   []byte // client cert
+	key    []byte // client private key
+	cacert []byte // ca server cert
+
+}
+
+// NewX509 returns a new to an empty X509
+func NewX509() *X509 {
```

Comment:
The tests are all commented out, and this isn't used in the winrm package. Will you be using it in the juju code? If yes, please uncomment the tests and make sure they're working. If not, please delete the code.